docs/general/adapters.md

```diff
@@ -194,6 +194,43 @@ The user could pass in `include=**`.
 
 We recommend filtering any user-supplied includes appropriately.
 
+#### Include Parameters
+
+You may wish to include related resources based on the `include` request
+parameter (See [fetching_includes](http://jsonapi.org/format/#fetching-includes). You can opt-in to this pattern:
+
+```ruby
+class PostSerializer < ActiveModel::Serializer
+  associations_via_include_param(true)
```

Comment:
I kind of have a hard time following what the changes are trying to do in the PR mostly because of the naming, I think

- What does `_associations_via_include_param` connote?
- What's the difference between `include_directive` and `current_include_directive`?
- What's the difference between `current_include_directive` and `full_include_directive`
- Why are there two include directives now?
- Why is there a param include_directive introduced that defaults to `{}` when no arg passed?
- What does `load_data` connote? 
- What's the difference between `include_data` and `load_data`?
- Why is the consequence of `include_data?` a choice between `@_load_data.call` and `include_data_for`?